packages/web-components/src/components/va-banner/va-banner.tsx

```diff
@@ -0,0 +1,185 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  h,
+  Prop,
+  State,
+} from '@stencil/core';
+
+const DISMISSED_BANNERS_KEY = 'DISMISSED_BANNERS';
+@Component({
+  tag: 'va-banner',
+  styleUrl: 'va-banner.css',
+  shadow: true,
+})
+export class VaBanner {
+  @Element() el: HTMLElement;
+
+  /**
+   * If true, doesn't fire the CustomEvent which can be used for analytics tracking.
+   */
+  @Prop() disableAnalytics: boolean = false;
+
+  /**
+   * Enable the close functionality. The banner will be closed until storage is cleared.
+   * */
+  @Prop() showClose: boolean = false;
+
+  /**
+   * The headline of the banner.
+   * */
+  @Prop() headline: string;
+
+  /**
+   * The type of the banner. One of 'info', 'error', 'success', 'continue', or 'warning'. This affects both the icon of the AlertBox and the top border color.
+   * */
+  @Prop() type: string = 'info';
+
+  /**
+   * A boolean that when false makes it so that the banner does not render.
+   * */
+  @Prop() visible: boolean = true;
+
+  /**
+   * Enable sessionStorage for the Banner otherwise storage
+   * if showClose is enabled will default to localStorage
+   * */
+  @Prop() windowSession: boolean = false;
+
+  /**
+   * Keep track of locally dismissed Banners
+   * */
+  @State() dismissedBanners: Array<String> = [];
+
+  /**
+   * The event used to track usage of the component. This is emitted when an
+   * anchor link or the dismiss icon is clicked and disableAnalytics is not true.
+   */
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  private prepareBannerID = () => `${this.headline}:${this.el.innerHTML}`;
+
+  private dismiss = () => {
+    // Derive the current banner ID.
+    const currentBannerID = this.prepareBannerID();
+
+    // Escape early if the banner is already dismissed.
+    if (this.dismissedBanners.includes(currentBannerID)) {
+      return;
+    }
+
+    // Add the banner ID to the list of dismissed banners. If showClose enabled
+    if (this.showClose) {
+      // Derive the updated dismissed banners.
+      const updatedDismissedBanners = [
+        ...this.dismissedBanners,
+        currentBannerID,
+      ];
+
+      //   Set storage type
+      const storageType = this.windowSession
+        ? window.sessionStorage
+        : window.localStorage;
+
+      // Set banner dismissed in storage
+      storageType.setItem(
+        DISMISSED_BANNERS_KEY,
+        JSON.stringify(updatedDismissedBanners),
+      );
+
+      // Update dismissedBanners in state.
+      this.dismissedBanners = updatedDismissedBanners;
+
+      // Track the dismiss event in Google Analytics
+      if (!this.disableAnalytics) {
+        const detail = {
+          componentName: 'Banner',
```

Comment:
You are correct @bkjohnson.  We have been matching the web component name.

Reply:
@bsmartin-ep thank you I will update the component name. However a couple of questions:
1. What would you like the details of the Analytics event to be for dismiss? Right now it is clickLabel: "Dismiss Banner", headline: string, showClose: boolean, windowSession: boolean, type: string
2. We have a va-alert component nested inside our va-banner component. Therefore the analytics event for the va-alert that is enabled for when a link is clicked will fire inside of the va-banner component. Do we need a second analytics event that would be exclusive to the va-banner for when a link is clicked?

Reply:
@pdavies88 -

Apologies - missed these questions earlier.

1.  This is event `int-alert-box-close`, correct?  We're only looking for `alert-box-headline` in the dataLayer.  Which got me thinking [this setup file isn't working correctly](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/site-wide/component-library-analytics-setup.js).  We just [kebab case the component name](https://github.com/department-of-veterans-affairs/vets-website/blob/9e5c8399e9dbe100a88ee22899ed9d94070538c3/src/platform/site-wide/component-library-analytics-setup.js#L58) and prepend it to the dataLayer key.  Which won't work for us here, since we'll get `va-alert-headline`.  Want me to send in a PR allowing us to set an analytics prefix in that setup file?
2. No - unless there are new event parameters we need to add, I think deferring to the `va-alert` event sounds good to me.

Reply:
@bsmartin-ep thanks for the info yes the original event is `int-alert-box-close`. 

So correct me if I am wrong but I believe the event would become `va-banner-alert-box-headline` https://github.com/department-of-veterans-affairs/component-library/blob/caa96a470c90b18725ddc6c24c022c653974472b/packages/web-components/src/components/va-banner/va-banner.tsx#L103-L109

and then in https://github.com/department-of-veterans-affairs/vets-website/blob/9e5c8399e9dbe100a88ee22899ed9d94070538c3/src/platform/site-wide/component-library-analytics-setup.js#L9-L35

I would need to add in:
`'va-banner': [{ action: 'click', event: 'int-alert-box-close' }],`
and 
`'va-alert': [{ action: 'linkClick', event: 'nav-warning-alert-box-content-link-click' }],`
since they don't exist there yet

Reply:
>So correct me if I am wrong but I believe the event would become va-banner-alert-box-headline

@pdavies88 -

Unfortunately I don't think this will work for us. Our tagging system gives us some flexibility with regex-ing event _names_ but not parameters.  We would have to create a mess of fall-through default values in GTM.

I think our best bet is to move away from the kebab-casing of the component name and add in the GA event parameter prefix we want to the lookup table:

```js
'va-alert': [
     { action: 'linkClick', parameter_prefix: 'alert-box', event: 'nav-warning-alert-box-content-link-click' }
],
```

...or maybe slightly restructuring the array into an object...

```js
'va-alert': {
     parameter_prefix: 'alert-box',
     events: [{ action: 'linkClick', , event: 'nav-warning-alert-box-content-link-click' }]
},
```

Happy to send in a PR.  Or fix this in a separate concurrent issue (since your changes didn't cause this preexisting issue).